### PR TITLE
Rearrange getEndValue logic

### PR DESCRIPTION
**Pull request by susiwen8**
@zombieJ New fix based on our discussion.  Thanks for your advise.

### CODE REVIEW

File: tests/RangeCalendar.spec.jsx [hunk 1]
```diff
@@ -200,6 +200,15 @@ describe('RangeCalendar', () => {
...
   });
 
+  describe('calendar header', () => {
+    it('range date has selected, calendar header match date selected', () => {
+      const rangeCalendar = <RangeCalendar selectedValue={[moment('2000-09-03', format), moment('2000-10-28', format)]} timePicker={{}}/>;
+      const wrapper = mount(rangeCalendar);
+      expect(wrapper.find('.rc-calendar-month-select').at(0).text()).toEqual('Sep');
+      expect(wrapper.find('.rc-calendar-month-select').at(1).text()).toEqual('Oct');
+    });
+  });
```
Comment (zombieJ) [inline tests/RangeCalendar.spec.jsx]: Seems this test could be same start & end date but different time. We should mock the click operation to open time panel and check if start & end are the same date.

Comment (susiwen8) [reply tests/RangeCalendar.spec.jsx]: @zombieJ I have rewrite test case, please check

**Comment by zombieJ** [comment]
fix https://github.com/ant-design/ant-design/issues/16215

**Comment by afc163** [comment]
ant-design/ant-design#16215 is not fixed because antd still stuck in rc-calendar@9.12.x

**Comment by susiwen8** [comment]
@afc163 https://github.com/ant-design/ant-design/pull/16560

**Pull request merged**